protocol/v1/qbft/instance/pre_prepare.go

```diff
@@ -95,15 +47,17 @@ upon receiving a valid ⟨PRE-PREPARE, λi, ri, value⟩ message m from leader(
 */
 func (i *Instance) UponPrePrepareMsg() pipelines.SignedMessagePipeline {
 	return pipelines.WrapFunc("upon pre-prepare msg", func(signedMessage *specqbft.SignedMessage) error {
-		prepareMsg, err := signedMessage.Message.GetProposalData()
-		if err != nil {
-			return errors.Wrap(err, "failed to get prepare message")
+		newRound := signedMessage.Message.Round
+
+		// A future justified proposal should bump us into future round and reset timer
+		if signedMessage.Message.Round > i.State().GetRound() {
+			i.ResetRoundTimer() // TODO: make sure what is needed here is i.ResetRoundTimer(), not something else
```

Comment:
@nkryuchkov we can't bump round based on propose msg. 
what the case you tried to solve? 
if we getting higher round we need to fail the msg

Reply:
@nivBlox It's according to spec: https://github.com/bloxapp/ssv-spec/blob/main/qbft/proposal.go#L28. Should I do it differently?